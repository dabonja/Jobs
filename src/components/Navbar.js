import React from 'react';
import { Link } from "react-router-dom";

const Navbar = () => {
  return (
    <>
    <nav className="navbar navbar-expand-lg bg-dark" style={{fontFamily: 'Gill Sans, sans-serif',fontSize: '1.3rem'}}>
      <div className="container-fluid">
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarTogglerDemo01" aria-controls="navbarTogglerDemo01" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarTogglerDemo01">
          <h3 className="navbar-brand" style={{color: 'white',fontSize: '2rem'}}>Jobs</h3>
          <ul className="navbar-nav me-auto mb-2 mb-lg-0">
            <li className="nav-item">
              <Link className="nav-link active" aria-current="page" to="/">Home</Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link" to="/Artisans" >Artisans</Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link " tabIndex="-1" to="/ArtisanForm">Sign up for advertisment</Link>
            </li>
            <li className="nav-item">
              <Link className="nav-link disabled" to="/" tabIndex="-1" aria-disabled="true" >Sign your company for advertisment</Link>
            </li>
          </ul>
          <form className="d-flex">
            <Link className="nav-link disabled" to="/" tabIndex="-1" aria-disabled="true">Login</Link>
          </form>
        </div>
      </div>
    </nav>
    </>
  )
}

export default Navbar