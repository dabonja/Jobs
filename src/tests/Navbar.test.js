import Navbar from '../components/Navbar';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders Navbar component correctly', () => {

    const tree = renderer.create(
        <Router>
            <Navbar />
        </Router>)
        .toJSON();
    expect(tree).toMatchSnapshot()
})