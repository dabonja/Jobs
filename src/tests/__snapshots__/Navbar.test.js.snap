// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Navbar component correctly 1`] = `
<nav
  className="navbar navbar-expand-lg bg-dark"
  style={
    Object {
      "fontFamily": "Gill Sans, sans-serif",
      "fontSize": "1.3rem",
    }
  }
>
  <div
    className="container-fluid"
  >
    <button
      aria-controls="navbarTogglerDemo01"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarTogglerDemo01"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarTogglerDemo01"
    >
      <h3
        className="navbar-brand"
        style={
          Object {
            "color": "white",
            "fontSize": "2rem",
          }
        }
      >
        Jobs
      </h3>
      <ul
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Artisans"
            onClick={[Function]}
          >
            Artisans
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link "
            href="/ArtisanForm"
            onClick={[Function]}
            tabIndex="-1"
          >
            Sign up for advertisment
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-disabled="true"
            className="nav-link disabled"
            href="/"
            onClick={[Function]}
            tabIndex="-1"
          >
            Sign your company for advertisment
          </a>
        </li>
      </ul>
      <form
        className="d-flex"
      >
        <a
          aria-disabled="true"
          className="nav-link disabled"
          href="/"
          onClick={[Function]}
          tabIndex="-1"
        >
          Login
        </a>
      </form>
    </div>
  </div>
</nav>
`;
